docs/src/content/docs/core/define-plugin.mdx

```diff
@@ -0,0 +1,68 @@
+---
+title: definePlugin
+description: Allows defining a type-safe plugin that can be used in defineIntegration.
+---
+import Disabled from "~/components/Disabled.astro"
+import { LinkCard } from '@astrojs/starlight/components';
+
+Allows defining a type-safe plugin that can be used in `defineIntegration`.
+
+:::caution
+This utility is for advanced use-cases. It's useful for developers that want to create
+3rd party plugins for Astro Integration Kit, or if you want to learn a little bit more
+about the internals.
+:::
+
+```ts title="package/plugins/add-vite-plugin.ts"
+import { definePlugin } from "../core/define-plugin.js";
+import { addVitePlugin } from "../utilities/add-vite-plugin.js";
+import type { Plugin as VitePlugin } from "vite"
+
+export const addVitePluginPlugin = definePlugin({
+	name: "addVitePlugin",
+	hook: "astro:config:setup",
+	implementation:
+		({ updateConfig }) =>
+		(plugin: VitePlugin) =>
+			addVitePlugin({ plugin, updateConfig }),
+});
+```
+
+You can then use it in `defineIntegration`:
+
+```ts title="my-integration/index.ts" "addVitePluginPlugin"
+import { defineIntegration } from "astro-integration-kit";
+import { addVitePluginPlugin } from "../package/plugins/add-vite-plugin.js";
+
+export default defineIntegration({
+    name: "my-integration",
+    plugins: [addVitePluginPlugin],
```

Comment:
Yeah so actually I mentioned this further down, but not suffixing the plugins with the word "Plugin" would mean we wouldn't have `addVitePluginPlugin` 😂